From 802b1a530b8802ba2c45667f4829ba24feb882f7 Mon Sep 17 00:00:00 2001
From: Ovidiu Panait <ovidiu.panait@windriver.com>
Date: Fri, 16 Feb 2018 21:06:51 +0200
Subject: [PATCH] net: ethernet: xilinx: bug fix for missing phydev var

Fixes the following compilation error when CONFIG_XILINX_AXI_EMAC is
enabled:
...
drivers/net/ethernet/xilinx/xilinx_axienet_main.c:1513:6: error:
phydev' undeclared (first use in this function)
|   if (phydev)
...

This fix is specific to the Windriver xilinx-zynq kernel (the compilation
issue was introduced by commit 0031e853a0496a96fc30c080c63862961ecced34).

Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_axienet_main.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index 25b7f61..14529df 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -1510,8 +1510,8 @@ err_rx_irq:
 	free_irq(lp->tx_irq, ndev);
 err_tx_irq:
 	napi_disable(&lp->napi);
-	if (phydev)
-		phy_disconnect(phydev);
+	if (lp->phy_dev)
+		phy_disconnect(lp->phy_dev);
 	tasklet_kill(&lp->dma_err_tasklet);
 	dev_err(lp->dev, "request_irq() failed\n");
 	return ret;
-- 
1.7.5.4

